pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -472,13 +474,15 @@ public SuccessResponse reloadSegmentDeprecated2(
   @ApiOperation(value = "Reload all segments", notes = "Reload all segments")
   public SuccessResponse reloadAllSegments(
       @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
-      @ApiParam(value = "OFFLINE|REALTIME") @QueryParam("type") String tableTypeStr) {
+      @ApiParam(value = "OFFLINE|REALTIME") @QueryParam("type") String tableTypeStr,
+      @ApiParam(value = "Whether to force server to download segment") @QueryParam("forceDownload") @DefaultValue("false") boolean forceDownload) {
```

Comment:
When force download is enabled, we should check the table type and replace `null` with `OFFLINE` to avoid reloading the realtime table

Reply:
the PinotHelixResourceManager.reloadSegment and reloadAllSegment check on table type, and right now only allow OFFLINE table to continue.

Reply:
The scenario we want to avoid is when the raw table name is provided, and no table type is specified. We don't want to trigger the force download reload on the realtime table, but only the offline table.

Reply:
If we want to remove indexes, then we would want to do this for realtime segments also, right? Not sure why you ask that it not be downloaded.

Reply:
The target of this feature is to clean obsolete indices on disk, and work for immutable segments from either realtime or offline table. But this PR works with OFFLINE table only as it reuses the addOrReplaceOfflineSegment() method, which assumes the segment is from OFFLINE table.

However, Jackie has opened a RP that removes limitations on ZKMetadata and would allow me to converge the code path of segment downloading for both Offline and Realtime table, and make this feature work with table types (sth I plan to work on as next)